vm/src/obj/objbytearray.rs

```diff
@@ -563,6 +565,17 @@ impl PyByteArray {
     fn decode(zelf: PyRef<Self>, args: DecodeArgs, vm: &VirtualMachine) -> PyResult<PyStringRef> {
         bytes_decode(zelf.into_object(), args, vm)
     }
+
+    #[pymethod(magic)]
+    fn reduce_ex(zelf: PyRef<Self>, _proto: usize, vm: &VirtualMachine) -> (PyClassRef, PyTuple) {
+        Self::reduce(zelf, vm)
+    }
+
+    #[pymethod(magic)]
+    fn reduce(zelf: PyRef<Self>, vm: &VirtualMachine) -> (PyClassRef, PyTuple) {
+        let bytes = PyBytes::from(zelf.borrow_value().elements.clone()).into_pyobject(vm);
+        (Self::class(vm), PyTuple::from(vec![bytes]))
+    }
```

Comment:
Unlike CPython, I find out that protocol does not need treat different than we get the right result. I think CPython use it for compatibility with python2.

Reply:
I think python3 still support lower proto versions.